TSC: Cannot find name 'mgrs'. Cannot find module 'geotiff' or its corresponding type declarations.

https://github.com/proj4js/proj4js/blob/38d1c4b02d0fa8251b92c2e5d27ebba295f2b238/lib/index.js#L8C19-L8C23

```sh
17:30:04  node_modules/proj4/dist/lib/index.d.ts(15,18): error TS2304: Cannot find name 'mgrs'.
17:30:04  node_modules/proj4/dist/lib/nadgrid.d.ts(15,59): error TS2307: Cannot find module 'geotiff' or its corresponding type declarations.
```

--- a/lib/index.js
+++ b/lib/index.js
@@ -9,8 +9,15 @@ import mgrs from 'mgrs';
 import includedProjections from '../projs';
 
 /**
+ * @typedef {Object} Mgrs
+ * @property {(lonlat: [number, number]) => string} forward
+ * @property {(mgrsString: string) => [number, number, number, number]} inverse
+ * @property {(mgrsString: string) => [number, number]} toPoint
+ */
+
+/**
  * @template {import('./core').TemplateCoordinates} T
- * @type {core<T> & {defaultDatum: string, Proj: typeof Proj, WGS84: Proj, Point: typeof Point, toPoint: typeof common, defs: typeof defs, nadgrid: typeof nadgrid, transform: typeof transform, mgrs: typeof mgrs, version: string}}
+ * @type {core<T> & {defaultDatum: string, Proj: typeof Proj, WGS84: Proj, Point: typeof Point, toPoint: typeof common, defs: typeof defs, nadgrid: typeof nadgrid, transform: typeof transform, mgrs: Mgrs, version: string}}
  */
 const proj4 = Object.assign(core, {
   defaultDatum: 'WGS84',

--- a/package.json
+++ b/package.json
@@ -36,7 +36,6 @@
     "@stylistic/eslint-plugin": "^4.2.0",
     "chai": "^5.0.0",
     "eslint": "^9.25.1",
-    "geotiff": "^2.1.3",
     "grunt": "^1.6.1",
     "grunt-cli": "^1.4.3",
     "mocha": "^10.2.0",
@@ -48,5 +47,8 @@
   "dependencies": {
     "mgrs": "1.0.0",
     "wkt-parser": "^1.5.1"
+  },
+  "peerDependencies": {
+    "geotiff": "*"
   }
 }

